crate/server/src/core/operations/delete_attribute.rs

```diff
@@ -0,0 +1,128 @@
+use cosmian_kmip::kmip::{
+    kmip_operations::{DeleteAttribute, DeleteAttributeResponse},
+    kmip_types::{Attribute, AttributeReference, Tag, UniqueIdentifier},
+};
+use cosmian_kms_client::access::ObjectOperationType;
+use tracing::trace;
+
+use crate::{
+    core::{extra_database_params::ExtraDatabaseParams, KMS},
+    database::retrieve_object_for_operation,
+    error::KmsError,
+    result::{KResult, KResultHelper},
+};
+
+pub(crate) async fn delete_attribute(
+    kms: &KMS,
+    request: DeleteAttribute,
+    user: &str,
+    params: Option<&ExtraDatabaseParams>,
+) -> KResult<DeleteAttributeResponse> {
+    trace!("Delete attribute: {}", serde_json::to_string(&request)?);
+
+    // there must be an identifier
+    let uid_or_tags = request
+        .unique_identifier
+        .as_ref()
+        .ok_or(KmsError::UnsupportedPlaceholder)?
+        .as_str()
+        .context("Delete Attribute: the unique identifier must be a string")?;
+
+    let mut owm = retrieve_object_for_operation(
+        uid_or_tags,
+        ObjectOperationType::GetAttributes,
+        kms,
+        user,
+        params,
+    )
+    .await?;
+    trace!(
+        "Delete Attribute: Retrieved object for: {:?}",
+        serde_json::to_string(&owm)
+    );
+
+    let mut attributes = owm.attributes;
+
+    if let Some(attribute) = request.current_attribute {
+        match attribute {
+            Attribute::ActivationDate(_) => {
+                attributes.activation_date = None;
+            }
+            Attribute::CryptographicAlgorithm(_) => {
+                attributes.cryptographic_algorithm = None;
+            }
+            Attribute::CryptographicLength(_) => {
+                attributes.cryptographic_length = None;
+            }
+            Attribute::CryptographicParameters(_) => {
+                attributes.cryptographic_parameters = None;
+            }
+            Attribute::CryptographicDomainParameters(_) => {
+                attributes.cryptographic_domain_parameters = None;
+            }
+            Attribute::CryptographicUsageMask(_) => {
+                attributes.cryptographic_usage_mask = None;
+            }
+            Attribute::Links(_) => {
+                attributes.link = None;
+            }
+            Attribute::VendorAttributes(_) => {
+                attributes.vendor_attributes = None;
+            }
+            x @ Attribute::State(_) => {
+                return Err(KmsError::NotSupported(format!(
+                    "Delete Attribute on {x:?} not supported"
+                )));
+            }
+        }
+    };
+
+    if let Some(attribute_references) = request.attribute_references {
```

Comment:
I don't get what `attribute_references` are used for - Can you explain a bit more ? :)

Reply:
the `attribute_references` help to identify the type of attribute to be deleted. There is a long list a `Tag` that helps to identify attributes:
```
pub enum Tag {
    #[value(name = "ActivationDate")]
    ActivationDate = 0x42_0001,
    #[value(name = "ApplicationData")]
    ApplicationData = 0x42_0002,
...
```
in kmip_types.rs